src/net.cpp

```diff
@@ -361,11 +361,38 @@ CNode* CConnman::FindNode(const CService& addr, std::optional<bool> inbound)
     return nullptr;
 }
 
-bool CConnman::AlreadyConnectedToAddress(const CAddress& addr)
+bool CConnman::AlreadyConnectedToAddress(const CAddress& addr, ConnectionType conn_type, bool log)
```

Comment:
commit: `p2p, bugfix: correctly detect connected peers in AlreadyConnectedToAddress()`
Could you list some examples of scenarios that are not working correctly currently and are improved by this? I'm trying to understand the impacts of this commit better.